src/components/ResourceFilter.jsx

```diff
@@ -1,34 +1,37 @@
-import * as React from "react"
 import FilterButton from "./FilterButton"
 import ResourceIcon from "../assets/Resource.svg"
 import ResourceIconDeactivated from "../assets/ResourceDeactivated.svg"
-import { useSlug } from "~/context/slug"
-import { useData } from "~/context/data"
+import { useResources } from "~/hooks/useResources"
+import Skeleton from "react-loading-skeleton"
 
-export default function ResourceFilter() {
-  const { location, resource } = useSlug()
-  const { resources: data } = useData()
+export default function ResourceFilter({ city, resource }) {
+  const { data, error } = useResources()
 
-  if (!data) return null
+  if (error) return <div>failed to load</div>
+  if (!data) return <Skeleton height={128} />
 
-  if (location)
+  const resources = Object.keys(data)
```

Comment:
Please follow your own conventions, use either this or useEffect method in both LocationFilter and ResourceFilter.

Move this Object to Array logic inside useResources since an Array is what the component consumes.